coredocument/coredocument.go

```diff
@@ -295,3 +320,30 @@ func CreateProofs(dataTree *proofs.DocumentTree, coreDoc *coredocumentpb.CoreDoc
 
 	return proofs, nil
 }
+
+func fetchUniqueCollaborators(oldCollabs [][]byte, newCollabs []string) (ids []identity.CentID, err error) {
+	ocsm := make(map[string]struct{})
```

Comment:
Maybe we can move this logic as a util to create "sets" of strings. I think it will be useful in other places.

Reply:
We can move this when we need it maybe ?

Reply:
I wouldn't touch the collaborators stuff too much. We will get rid of that part of the document when implementing write ACL. Then it will be completely replaced by ACLs.